### ISSUE TITLE
SendAsync(msg).Wait(1000) always "false" on custom server

### ISSUE THREAD

**Issue by xarthurx** [jchristn/WatsonWebsocket#96]
I'm buiding a communication between c# (WatsonWebsocket) and cpp (Websocket server).

When calling `SendAsync(msg).Wait(1000)`, it seems the result is always `false` though I can guarantee the server has received the message.

As I'm not using `WatsonWebsocket` to build the server, I assume I need to send something back from my custom server so that the `WastonClient` knows the message is sent correctly?
What should I send back?

https://github.com/jchristn/WatsonWebsocket/issues/27

**Comment by xarthurx** [comment]
If I try to use `SendAsync(msg).Result` it takes forever to return the `bool` result... Never get it.

**Comment by jchristn** [comment]
You shouldn't have to send anything back from the server.  What happens if you do it without ```.Wait(1000)```?

**Comment by xarthurx** [comment]
It can send data to the server.
But if the server is not started, the program hangs forever.

Besides, I need sth like this:
![image](https://user-images.githubusercontent.com/1921878/182210858-b3c03187-3eb7-468f-b85e-8566873e0147.png)

**Comment by jchristn** [comment]
You could always do ```bool success = _Client.SendAsync(curText);``` followed by ```if (!success)```

**Comment by xarthurx** [comment]
> You could always do `bool success = _Client.SendAsync(curText);` followed by `if (!success)`

I cannot. Otherwise I would not start this issue.
![image](https://user-images.githubusercontent.com/1921878/182223971-fe6eae84-6620-46b1-b05e-58290ff589d6.png)

**Comment by xarthurx** [comment]
check this:

https://stackoverflow.com/questions/45138268/how-to-get-bool-result-from-async-taskbool-function-in-c-sharp-error-cannot

**Comment by xarthurx** [comment]
I think we're also missing functions like `client.IsRunning` to check the status of the client?
May I ask how should I check if a client is connected to to server?

**Comment by xarthurx** [comment]
OK, I got it to work by 
`bool success = await _Client.SendAsync(curText);`
And found `ServerConnected` properties.

**Comment by jchristn** [comment]
Right, sorry, I should have said ```bool success = await ...```, good catch!
